homeassistant/components/ecovacs/vacuum.py

```diff
@@ -79,60 +92,72 @@ def on_error(self, error):
 
         self.hass.bus.fire(
             "ecovacs_error", {"entity_id": self.entity_id, "error": error}
         )
         self.schedule_update_ha_state()
 
     @property
-    def unique_id(self) -> str:
-        """Return an unique ID."""
-        return self.device.vacuum.get("did")
-
-    @property
-    def is_on(self) -> bool:
-        """Return true if vacuum is currently cleaning."""
-        return self.device.is_cleaning
+    def state(self) -> str | None:
+        """Return the state of the vacuum cleaner."""
+        if self.error is not None:
+            return STATE_ERROR
 
-    @property
-    def is_charging(self) -> bool:
-        """Return true if vacuum is currently charging."""
-        return self.device.is_charging
+        if self.device.is_cleaning:
+            return STATE_CLEANING
 
-    @property
-    def status(self) -> str | None:
-        """Return the status of the vacuum cleaner."""
-        return self.device.vacuum_status
+        if self.device.is_charging:
+            return STATE_DOCKED
 
-    def return_to_base(self, **kwargs: Any) -> None:
-        """Set the vacuum cleaner to return to the dock."""
+        if self.device.vacuum_status == sucks.CLEAN_MODE_STOP:
+            return STATE_PAUSED
 
-        self.device.run(sucks.Charge())
+        if self.device.vacuum_status == sucks.CHARGE_MODE_RETURNING:
+            return STATE_RETURNING
 
-    @property
-    def battery_icon(self) -> str:
-        """Return the battery icon for the vacuum cleaner."""
-        return icon_for_battery_level(
-            battery_level=self.battery_level, charging=self.is_charging
-        )
+        return STATE_IDLE
 
     @property
     def battery_level(self) -> int | None:
         """Return the battery level of the vacuum cleaner."""
         if self.device.battery_status is not None:
             return self.device.battery_status * 100
 
-        return super().battery_level
+        return None
+
+    @property
+    def battery_icon(self) -> str:
+        """Return the battery icon for the vacuum cleaner."""
+        return icon_for_battery_level(
+            battery_level=self.battery_level, charging=self.device.is_charging
+        )
 
     @property
     def fan_speed(self) -> str | None:
         """Return the fan speed of the vacuum cleaner."""
         return self.device.fan_speed
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the device-specific state attributes of this vacuum."""
+        data: dict[str, Any] = {}
+        data[ATTR_ERROR] = self.error
+
+        for key, val in self.device.components.items():
+            attr_name = ATTR_COMPONENT_PREFIX + key
+            data[attr_name] = int(val * 100)
+
+        return data
+
+    def return_to_base(self, **kwargs: Any) -> None:
+        """Set the vacuum cleaner to return to the dock."""
+
+        self.device.run(sucks.Charge())
+
     def turn_on(self, **kwargs: Any) -> None:
```

Comment:
```suggestion
    def start(self, **kwargs: Any) -> None:
```

Also, rename `turn_off` to `stop` and `stop` to `pause`

Reply:
what is the difference between `pause` and `stop`? from documentation it sounds like the same, because both does just stop the vacuum, but do not "return to base" 🤔